diff --git a/src/shim/platform.h b/src/shim/platform.h
index 722b1f0..8b2e336 100644
--- a/src/shim/platform.h
+++ b/src/shim/platform.h
@@ -215,14 +215,15 @@ protected:
   bool
   load_bo_info(uint32_t key, bo_info& info) const;
 
-private:
-  std::shared_ptr<const drv> m_driver;
-
+protected:
   // Supposed to be set once and used till object is destroyed.
   // No locking protection here. Caller should make sure there is no race.
   mutable int m_dev_fd = -1;
   mutable std::string m_sysfs_root;
 
+private:
+  std::shared_ptr<const drv> m_driver;
+
   // Proces-wide drm_bo map to allow retrieving the bo by handle.
   // If a bo is exported, then imported within the same process, we will
   // get back the same bo handle. We can, then, retrieve the bo back
